ioctl/test_ioctl.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    Phoenix-RTOS standard library functions tests
+ *    HEADER:
+ *    - ioctl.h
+ *    TESTED:
+ *    - ioctl()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <phoenix/ioctl.h>
+#include <sys/msg.h>
+#include <signal.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <posix/utils.h>
+#include <unity_fixture.h>
+
+static uint32_t port;
+static int fd, dd = 5;
+static volatile int loop_flag = 1;
+
+typedef uint8_t test_ioctl_buf[64];
+
+#define MAX_FAIL 10
+#define PATH_TF  "ioctl_testFile"
+#define PATH_REG "ioctl_testRegularFile"
+#define DEV      "/dev/test"
+/* T stands for test, do not confuse with tty */
+#define TEST_GRP ('T')
+
+#define TEST_IOCTL_SIG       _IOC(IOC_VOID, TEST_GRP, 0x01, 0)
+#define TEST_IOCTL_IN_VAL    _IOC(IOC_VOID, TEST_GRP, 0x02, sizeof(int32_t))
+#define TEST_IOCTL_IN        _IOC(IOC_IN, TEST_GRP, 0x03, sizeof(int32_t))
+#define TEST_IOCTL_IN_BIG    _IOC(IOC_IN, TEST_GRP, 0x04, sizeof(test_ioctl_buf))
+#define TEST_IOCTL_OUT       _IOC(IOC_OUT, TEST_GRP, 0x05, sizeof(int32_t))
+#define TEST_IOCTL_OUT_BIG   _IOC(IOC_OUT, TEST_GRP, 0x06, sizeof(test_ioctl_buf))
+#define TEST_IOCTL_INOUT     _IOC(IOC_INOUT, TEST_GRP, 0x07, sizeof(int32_t))
+#define TEST_IOCTL_INOUT_BIG _IOC(IOC_INOUT, TEST_GRP, 0x08, sizeof(test_ioctl_buf))
+
+
+static void *thread(void *arg)
+{
+	msg_t msg;
+	unsigned long int rid;
+	int32_t out, flag = 0x12345678;
```

Comment:
the `expected_flag_value` should be a global const, not repeated twice as a magic number in different places